src/common/utils/setup-helpers.js

```diff
@@ -123,8 +124,13 @@ function createKalturaPlayerContainer(targetId: string): string {
  * @returns {void}
  */
 function setStorageConfig(options: KPOptionsObject): void {
-  if (!options.disableUserCache && StorageManager.isLocalStorageAvailable() && StorageManager.hasStorage()) {
-    Utils.Object.mergeDeep(options, StorageManager.getStorageConfig());
+  if (!options.disableUserCache) {
+    if (LocalStorageManager.isLocalStorageAvailable() && LocalStorageManager.hasStorage()) {
+      Utils.Object.mergeDeep(options, LocalStorageManager.getStorageConfig());
+    }
+    if (SessionStorageManager.isSessionStorageAvailable() && SessionStorageManager.hasStorage()) {
+      Utils.Object.mergeDeep(options, SessionStorageManager.getStorageConfig());
```

Comment:
I think setStorageConfig should also be by type. otherwise you're setting the same config into all storage managers, while currently its only relevant to local storage manager

Reply:
why it is setting the same config into all storage managers?
we first get the storage from local storage and merge it into options obj, then get the storage from session storage and merge it into options.

Reply:
yeah, you're right